operator/pkg/controller/clusterconfig/controller.go

```diff
@@ -114,26 +116,58 @@ func (r *ClusterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	// Starting here some change might be done on the cluster, so make sure to publish status update
-	defer r.client.Status().Update(ctx, &r.instance)
-
-	for _, rec := range []func(context.Context, logr.Logger) utils.SubReconcilerResult{
-		r.reconcileVirtualIPPools,
-	} {
-		if result := rec(ctx, reqLogger); result.ShouldReturn() {
-			r.instance.SetReadyCondition(metav1.ConditionFalse, "Reconciling", "Reconciliation in progress")
-			return result.GetResult()
-		}
+	defer r.client.Status().Update(ctx, instance)
+
+	subReconcilers := map[string]interface {
+		Load(*utils.ObjectHandler, logr.Logger, *metalk8sscalitycomv1alpha1.ClusterConfig)
+		GetLogger() logr.Logger
+		Reconcile(context.Context) utils.SubReconcilerResult
+	}{
+		"WorkloadPlaneVirtualIPPool": &workloadplane.VirtualIPPoolReconciler{},
 	}
 
-	r.instance.SetReadyCondition(metav1.ConditionTrue, "Ready", "Everything good")
+	// Load all sub reconcilers
+	for key, subrec := range subReconcilers {
+		logger := reqLogger.WithValues("SubReconciler", key)
+		handler := utils.NewObjectHandler(instance, r.client, r.scheme, r.recorder, logger, componentName, appName)
+		subrec.Load(handler, logger, instance)
+	}
 
-	return utils.EndReconciliation()
-}
+	var wg sync.WaitGroup
+	var mutex sync.Mutex
+	result := utils.NothingToDo()
 
-func (r *ClusterConfigReconciler) sendEvent(status metav1.ConditionStatus, reason string, message string) {
-	eventType := corev1.EventTypeNormal
-	if status == metav1.ConditionUnknown {
-		eventType = corev1.EventTypeWarning
+	// Run all sub reconcilers
+	for key := range subReconcilers {
+		wg.Add(1)
+		go func(key string) {
+			defer wg.Done()
+
+			subrec := subReconcilers[key]
+
+			logger := subrec.GetLogger()
+			logger.Info("reconciling: START")
+			defer logger.Info("reconciling: STOP")
+
+			if res := subrec.Reconcile(ctx); res.ShouldReturn() {
+				mutex.Lock()
+				if res.IsSuperior(result) {
+					result = res
+				}
+				mutex.Unlock()
+			}
+		}(key)
+	}
```

Comment:
LGTM, but would be nice to have this piece of logic (orchestrating subreconcilers) in a separate function/package, so we can test it easily.

Reply:
You mean the full "for" loop, yes ok we could